x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduler.java

```diff
@@ -273,4 +277,11 @@ public void deregisterTransform(String transformId) {
     List<TransformScheduledTask> getTransformScheduledTasks() {
         return scheduledTasks.listScheduledTasks();
     }
+
+    private TimeValue getFrequency(TimeValue frequency) {
+        if (frequency == null) {
+            frequency = Transform.DEFAULT_TRANSFORM_FREQUENCY;
```

Comment:
Since you are implementing the defaulting logic here, I think you could get rid of similar logic in `TransformScheduledTask`'s constructor and put an assertion in that constructor that `frequency` is not null.

Reply:
fixed